2048.py

```diff
@@ -0,0 +1,223 @@
+        while not_sorted_move_left_and_up(row):
+            left_and_up_sort(row)
+    return board
+
+
+def move_up(board):
+    for column in get_columns(board) :
+        while not_sorted_move_left_and_up(column):
+            left_and_up_sort(column)
+    return board
+
+
+def not_sorted_move_right_and_down(row):
+    for index in range(len(row) - 1, 0, -1):
+        while row[index].value == 0 and row[index - 1].value > 0:
+            return True
+    return False
+
+
+def right_and_down_sort(array):
+    for index in range(len(array) - 1, 0, -1):
+        if array[index].value == 0 and array[index - 1].value > 0:
+            temp_index = array[index].value
+            temp_index_minus_one = array[index - 1].value
+            array[index - 1].value = temp_index
+            array[index].value = temp_index_minus_one
+
+
+def move_right(board):
+    for row in board:
+        while not_sorted_move_right_and_down(row):
+            right_and_down_sort(row)
+    return board
+
+
+def move_down(board):
+    for column in get_columns(board):
+        while not_sorted_move_right_and_down(column):
+            right_and_down_sort(column)
+    return board
+
+
+def right_or_down_sum(array):
+    for index in range(0, len(array) - 1):
+        if array[index].value == array[index + 1].value:
+            array[index + 1].value += array[index].value
+            array[index].value = 0
+
+
+def sum_right(board):
+    for row in board:
+        right_or_down_sum(row)
+    return board
+
+
+def sum_down(board):
+    for column in get_columns(board):
+        right_or_down_sum(column)
+    return board
+
+
+def left_or_up_sum(array):
+    for index in range (len(array) -1, 0, -1):
+        if array[index].value == array[index - 1].value:
+            array[index -1].value += array[index].value
+            array[index].value = 0
+
+
+def sum_left(board):
+    for row in board:
+        left_or_up_sum(row)
+    return board
+
+
+def sum_up(board):
+    for column in get_columns(board):
+        left_or_up_sum(column)
+    return board
+
+
+def get_fields_with_zero_value(board):
+    fields_with_zero_value = []
+    for row in board:
+        for field in row:
+            if field.value == 0:
+                fields_with_zero_value.append(field)
+    return fields_with_zero_value
+
+def place_next_element_on_board(board):
+    next_field = random.choice(get_fields_with_zero_value(board))
+    next_field.value = 2
+
+
+def game_over(board):
+    winning_number = 2048
+    for row in board:
+        if winning_number in row:
+            print("You did it!")
+            return True
+    if len(get_fields_with_zero_value(board)) > 0:
+        return False
+    if len(get_fields_with_zero_value(board)) == 0:
+        print("YOU LOST!")
+        return True
+
+
+def valid_players_move(board):
+    players_move = str(input("Next move: ").lower())
+    if players_move == "a":
+        move_left(board)
+        sum_left(board)
+        move_left(board)
```

Comment:
hmm... If I remember our design of this method it didn't look like this implementation :D Check your notes!